File: python/stubSomeValues/birthdayservice.py
class BirthdayService(object):

    def __init__(self, client_repository):
        self.client_repository = client_repository

    def greeting(self, name):
        if self.client_repository.birthdayIsTodayFor(name):
            greeting = 'Happy birthday ' + name + '!'
        else:
            greeting = ''

        return greeting

File: php/src/stub/BirthdayService.php
<?php

namespace mock_and_stub_introduction\stub;

class BirthdayService {
    
    var $clientRepository;
    
    public function __construct($clientRepository) {
        $this->clientRepository = $clientRepository;
    }
    
    public function greetingFor($name) {
        $greeting = "";
        if ($this->clientRepository->birthDayIsTodayFor($name)) {
            $greeting = "Happy birthday $name!";
        }
        else {
            $greeting = "Good morning $name.";
        }
        return $greeting;
    }
    
}

?>
